app/services/cloud_topology_service.rb

```diff
@@ -0,0 +1,74 @@
+class CloudTopologyService < TopologyService
+  include UiServiceMixin
+
+  @provider_class = ManageIQ::Providers::CloudManager
+
+  def entity_type(entity)
+    entity.class.name.demodulize
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    included_relations = [
```

Comment:
for the format question: the format is ok, the last hash doesn't need {}, this is just passed as an argument for rails .includes